Resources/public/js/views/serverside/ez-roleserversideview.js

```diff
@@ -60,26 +63,86 @@ YUI.add('ez-roleserversideview', function (Y) {
         },
 
         /**
-         * Changes the state of the provided assign button to be *loading* and
+         * tap event handler for policy limitation on location ("Node"). It launches the
+         * universal discovery widget so that the user can pick a location.
+         *
+         * @method _pickLocationLimitation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _pickLocationLimitation: function (e) {
+            var button = e.target,
+                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button);
+
+            e.preventDefault();
+            this._uiSetUDWButtonLoading(button);
+            this.fire('contentDiscover', {
+                config: {
+                    title: button.getAttribute('data-universaldiscovery-title'),
+                    cancelDiscoverHandler: unsetLoading,
+                    multiple: true,
+                    contentDiscoveredHandler: Y.bind(this._setLocationLimitation, this, button),
+                },
+            });
+        },
+
+        /**
+         * Puts picked location id into location limitation input. Input is selected by the selector
+         * provided in the `data-location-input-selector` attribute of the button, for example
+         * <button data-location-input-selector="#id_of_input"></button>
+         *
+         * @method _setLocationLimitation
+         * @protected
+         * @param {Y.Node} button
+         * @param {EventFacade} e
+         */
+        _setLocationLimitation: function (button, e) {
+            var locationInput = this.get('container').one(button.getAttribute('data-location-input-selector')),
+                locationIdArray = [],
+                selectedLocation = this.get('container').one(button.getAttribute('data-selected-location-selector')),
+                selectedLocationList = selectedLocation.get('children').item(0);
+
+            while(selectedLocationList.firstChild) {
+                selectedLocationList.removeChild(selectedLocationList.firstChild);
+            }
+
+            Y.Array.each(e.selection, function (struct) {
+                var listItem = Y.config.doc.createElement('li');
```

Comment:
@sunpietro notice: I wrote *can* not *should* meaning it's not a strong requirement. The pros of using the Node API are the consistency and (maybe) a nicer API.
For `Y.config.doc` over `document`, this might be useful in tests in some cases when you precisely want to check what is done on the document but here it does not really matter as we can check the end result easily.